### PR TITLE
{data,tools}[gompi/2022a] CDO v2.0.6, ecCodes v2.27.0

### PR DESCRIPTION
**Pull request by maxim-masterov**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/c/CDO/CDO-2.0.6-gompi-2022a.eb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+# Use dependencies from EasyBuild
+configopts += "--with-curl=$EBROOTCURL --with-eccodes=$EBROOTECCODES --with-fftw3 --with-hdf5=$EBROOTHDF5 "
+configopts += "--with-netcdf=$EBROOTNETCDF --with-proj=$EBROOTPROJ --with-szlib=$EBROOTSZIP "
+configopts += "--with-udunits2=$EBROOTUDUNITS --with-util-linux-uuid=$EBROOTUTILMINLINUX "
+
+sanity_check_paths = {
+    'files': ['bin/cdo', 'lib/libcdi.a', 'lib/libcdi.%s' % SHLIB_EXT],
+    'dirs': ['include'],
+}
```
Comment (smoors) [inline easybuild/easyconfigs/c/CDO/CDO-2.0.6-gompi-2022a.eb:44]: @maxim-masterov please add back the `sanity_check_commands` from CDO-2.0.5-gompi-2021b.eb ?

Comment (maxim-masterov) [reply easybuild/easyconfigs/c/CDO/CDO-2.0.6-gompi-2022a.eb:44]: @smoors in the new version of CDO they start returning the `EXIT_FAILURE` code every time a user calls for a generic CLI option (e.g. `cdo -V`, or `cdo --help`), even if the command call succeeds. so, performing a sanity check with `cdo --version` fails with an exit code 1. therefore, I removed it from the easyconfig.

File: easybuild/easyconfigs/c/CDO/CDO-2.0.6-gompi-2022a.eb [hunk 2]
```diff
@@ -0,0 +1,46 @@
+easyblock = 'ConfigureMake'
```
[the remark this answers is not in the document]
Comment (maxim-masterov) [reply easybuild/easyconfigs/c/CDO/CDO-2.0.6-gompi-2022a.eb:6]: done

File: easybuild/easyconfigs/c/CDO/CDO-2.0.6-gompi-2022a.eb (hunk 1 above)
Comment (smoors) [reply easybuild/easyconfigs/c/CDO/CDO-2.0.6-gompi-2022a.eb:49]: @maxim-masterov you can work around that by grepping the output, for example in the breseq easyconfig:
`sanity_check_commands = ["breseq --help | grep 'Usage: breseq'"]`

and if the output is sent to stderr, you can redirect it, e.g.:
`sanity_check_commands = ["tmux --help 2>&1 | grep 'usage: tmux'"]`

Comment (maxim-masterov) [reply easybuild/easyconfigs/c/CDO/CDO-2.0.6-gompi-2022a.eb:49]: @smoors I see, thanks for the tip. done

File: easybuild/easyconfigs/c/CDO/CDO-2.0.6-gompi-2022a.eb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+configopts += "--with-curl=$EBROOTCURL --with-eccodes=$EBROOTECCODES --with-fftw3 --with-hdf5=$EBROOTHDF5 "
+configopts += "--with-netcdf=$EBROOTNETCDF --with-proj=$EBROOTPROJ --with-szlib=$EBROOTSZIP "
+configopts += "--with-udunits2=$EBROOTUDUNITS --with-util-linux-uuid=$EBROOTUTILMINLINUX "
+
+sanity_check_paths = {
+    'files': ['bin/cdo', 'lib/libcdi.a', 'lib/libcdi.%s' % SHLIB_EXT],
+    'dirs': ['include'],
+}
+
+sanity_check_commands = ["cdo --version  2>&1 | grep 'CDI library version : 2.0.6'"]
```
Comment (smoors) [inline easybuild/easyconfigs/c/CDO/CDO-2.0.6-gompi-2022a.eb:51]: thanks! one more suggestion:

```suggestion
sanity_check_commands = ["cdo --version  2>&1 | grep 'CDI library version : %(version)s'"]
```

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
```
-- Could NOT find AEC (missing: AEC_LIBRARY AEC_INCLUDE_DIR) 
CMake Error at cmake/ecbuild_log.cmake:190 (message):
  [1;35mCRITICAL - AEC library was not found.

  AEC (Adaptive Entropy Coding) provides the WMO GRIB CCSDS compression and
  decompression of data.  This is highly recommended from ecCodes >= 2.25.0

  To force the build without it, use -DENABLE_AEC=OFF[m
```

**Comment by smoors** [comment]
Test report by @smoors
**FAILED**
Build succeeded for 5 out of 7 (2 easyconfigs in total)
node304.hydra.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/03376e22e4a11947131126026fb6b2cb for a full test report.

**Comment by maxim-masterov** [comment]
@smoors Added `libaec` as a dependency

**Comment by smoors** [comment]
@boegelbot: please test @ generoso

**Comment by smoors** [comment]
@maxim-masterov the sanity check failed for me too because of the missing `bzip2` dep for `netCDF`. 
see https://github.com/easybuilders/easybuild-easyconfigs/pull/16450

after reinstalling `netCDF`, it worked for me.

are you hitting the same issue?

**Comment by smoors** [comment]
Test report by @smoors
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node302.hydra.os - Linux CentOS Linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 2.7.5
See https://gist.github.com/92d5260b6e2defec6fe27466c3efd3ec for a full test report.

**Review by smoors — APPROVED** [review]
lgtm

**Pull request merged**